File: fizzbuzz.scala
// Language:       Scala
// Web site:       http://www.scala-lang.org/
// Last tested on: Ubuntu 16.10
// Requires:       apt-get install scala openjdk-8-jdk-headless:amd64 openjdk-8-jre:amd64 openjdk-8-jre-headless:amd64
//                 (Uninstall openjdk-9 if necessary.)
//                 NOTE: The scalac and/or scala command invokes stty, causing it to hang if stdout is not a tty.

object fizzbuzz
{
    def main(args: Array[String]) = {
        for (i <- (1 to 100)) {
            if (i % 15 == 0) {
                println("FizzBuzz")
            }
            else if (i % 3 == 0) {
                println("Fizz")
            }
            else if (i % 5 == 0) {
                println("Buzz")
            }
            else {
                println(i)
            }
        }
    }
}

File: fizzbuzz.java
// Language:       Java
// Web site:       http://www.java.com/
// Last tested on: Ubuntu 22.04.01
// Requires:       apt-get install openjdk-11-jdk-headless:amd64

public class fizzbuzz {
    public static final void main( String args[] ) {
	for (int i = 1; i <= 100; i ++) {
            if (i % 15 == 0) {
                System.out.println("FizzBuzz");
            }
            else if (i % 3 == 0) {
                System.out.println("Fizz");
            }
            else if (i % 5 == 0) {
                System.out.println("Buzz");
            }
            else {
                System.out.println(i);
            }
	}
    }
}
